feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/tindergov/cards/TinderGovCardsViewModel.kt

```diff
@@ -71,38 +85,44 @@ class TinderGovCardsViewModel(
         .distinctUntilChangedBy { it.first.id to it.second to it.third }
         .shareInBackground()
 
-    val isCardDraggingAvailable = topReferendumWithDetails
-        .map { (_, summary, amount) ->
-            val summaryLoaded = summary?.isLoaded().orFalse()
-            val amountLoaded = amount?.isLoaded().orFalse()
-            summaryLoaded && amountLoaded
-        }
+    private var isVotingInProgress = MutableStateFlow(false)
+
+    val isCardDraggingAvailable = combine(isVotingInProgress, topReferendumWithDetails) { isVotingInProgress, (_, summary, amount) ->
+        val summaryLoaded = summary?.isLoaded().orFalse()
+        val amountLoaded = amount?.isLoaded().orFalse()
+        !isVotingInProgress && summaryLoaded && amountLoaded
+    }
+
+    val basketModelFlow = basketFlow
+        .map { items -> mapBasketModel(items.values.toList()) }
 
     init {
-        interactor.observeReferendaAvailableToVote(this)
-            .onEach { referenda ->
-                val currentReferendaIds = sortedReferendaFlow.value.map { it.id }.toSet()
-                val newReferenda = referenda.associateBy { it.id }
-                    .filter { it.key !in currentReferendaIds }
+        val referendaWithBasket = combineToPair(interactor.observeReferendaAvailableToVote(this), basketFlow)
+        referendaWithBasket.onEach { (referenda, basket) ->
+            val currentReferendaIds = sortedReferendaFlow.value.map { it.id }.toSet()
+            val newReferenda = referenda.associateBy { it.id }
+                .filter { it.key !in currentReferendaIds }
+                .filter { it.key !in basket }
```

Comment:
`associateBy` seems to be redundant here since we later iterating over map rather than utilizing it for constant time lookup

`referenda.filter { it.id !in currentReferendaIds && it.id !in basket }` has the same result

Finally, `basket` should be converted to set for constant time `contains`

Reply:
Okay, agree but It's not necessary to convert basket to set since basket is a map now. so `it.key !in basket` is working for `basket` keys